integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSpark2SqlParser.scala

```diff
@@ -562,7 +563,13 @@ class CarbonSpark2SqlParser extends CarbonDDLSqlParser {
   protected lazy val showCache: Parser[LogicalPlan] =
     SHOW ~> METACACHE ~> opt(ontable) <~ opt(";") ^^ {
       case table =>
-        CarbonShowCacheCommand(table)
+        CarbonShowCacheCommand(false, table)
+    }
+
+  protected lazy val showExecutorCache: Parser[LogicalPlan] =
+    SHOW ~> EXECUTOR ~> METACACHE ~> opt(ontable) <~ opt(";") ^^ {
```

Comment:
Combine both showCache and showexecutor Cache parsers

like (SHOW ~> opt(EXECUTOR) <~ METACACHE) ~ opt(ontable)

Reply:
done